src/daemon/collectd.c

```diff
@@ -487,11 +484,6 @@ int notify_systemd (void)
         su_size = sizeof (sa_family_t) + strlen (notifysocket);
         if (su_size > sizeof (su))
             su_size = sizeof (su);
-#else
-	ERROR ("Systemd socket uses Linux abstract namespace notation (\"%s\"), "
```

Comment:
@octo 543a57f8aff added a more global way of preventing non-linux systems from running through the systemd-specific code path. So this is indeed dead code. I agree with @rubenk it can be removed.